drivers/adc/adc_stm32.c

```diff
@@ -230,38 +230,16 @@ struct adc_stm32_cfg {
 static void adc_stm32_enable_dma_support(ADC_TypeDef *adc)
 {
 	/* Allow ADC to create DMA request and set to one-shot mode as implemented in HAL drivers */
-
-#if defined(CONFIG_SOC_SERIES_STM32H7X)
-
-#if defined(STM32H72X_ADC)
-	if (adc == ADC3) {
-		LL_ADC_REG_SetDMATransferMode(adc, LL_ADC3_REG_DMA_TRANSFER_LIMITED);
-	} else {
-		LL_ADC_REG_SetDataTransferMode(adc, LL_ADC_REG_DMA_TRANSFER_LIMITED);
-	}
-#elif defined(STM32H74X_ADC)
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_adc)
+	LL_ADC_REG_SetDMATransfer(adc, LL_ADC_REG_DMA_TRANSFER_UNLIMITED);
+#elif defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32U5X)
+	/* H72x ADC3 and U5 ADC4 are different from the rest, but this call works also for them,
+	 * so no need to call their specific function
+	 */
```

Comment:
Open: Does commenting LL shenanigans adds clarity to the reader here ?

Reply:
For a reader who would be aware that these ADCs are different (and I include myself in it), it mentions the fact that this is done on purpose, that the peculiarities of these ADC were taken into account.